code/game/machinery/doors/firedoor.dm

```diff
@@ -203,30 +198,59 @@
 		for(var/obj/machinery/firealarm/fire_panel in place.firealarms)
 			fire_panel.set_status()
 
-/obj/machinery/door/firedoor/proc/check_atmos(datum/source)
+/obj/machinery/door/firedoor/proc/refresh_shared_turfs(datum/source, list/leaving_members, list/joining_members)
+	SIGNAL_HANDLER
+	
+	for(var/obj/machinery/door/firedoor/firelock in my_area.firedoors)
+		firelock.check_adjacent_turfs()
```

Comment:
I think what you really want to be doing here is something like this, so any one firelock can have access to all the other firelock's lists of problems
```suggestion
/obj/machinery/door/firedoor/proc/refresh_shared_turfs(datum/merger/source, list/leaving_members, list/joining_members)
	SIGNAL_HANDLER
	if(source.origin != src) // We only want to do this once, this is a nice way of pulling that off
		return
	var/list/shared_problems = list()
	for(var/obj/machinery/door/firedoor/firelock in source.members)
		firelock.issue_turfs = shared_problems
		// check all the adjacent "valid" turfs around the firelock
```